namespace.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Gravitational Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rigging
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewNamespaceControl returns new instance of Namespace updater
+func NewNamespaceControl(
+	config NamespaceConfig) (*NamespaceControl, error) {
+	err := config.checkAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &NamespaceControl{
+		NamespaceConfig: config,
+		Entry: log.WithFields(log.Fields{
+			"Namespace": formatMeta(config.ObjectMeta),
+		}),
+	}, nil
+}
+
+// NamespaceConfig  is a Namespace control configuration
+type NamespaceConfig struct {
+	// Namespace is already parsed daemon set, will be used if present
```

Comment:
~~daemon set~~
Does not look like it will behave as described - i.e. `checkAndSetDefaults` seems to require this.